rtlmux.c

```diff
@@ -215,9 +215,10 @@ static void serverReadCB(struct bufferevent *bev, void *ctx) {
       slog(LOG_INFO, SLOG_INFO, "Connected to server.");
     } else { // Failed to receive the magic header
       slog(LOG_ERROR, SLOG_ERROR, "Failed to receive magic header from server.");
-      bufferevent_free(bev);
-      connectToServerSoon(ctx);
-      return;
+      //bufferevent_free(bev);
+      //connectToServerSoon(ctx);
+      //return;
+      serverInfo.state = SERVER_CONNECTED;
```

Comment:
I don't really see why this is being set to `SERVER_CONNECTED` when it hasn't actually happened. This would then go on using the buffer and context on a bad connection. It seems you'd likely just want to use the timeToExit = 2 here.

Reply:
More or less... yes!